### DISCUSSION TITLE
[ALG 101]針對Chrome DevTool中練習debugger的狀況

### DISCUSSION THREAD

**Discussion by IamTomIamQQ** [Lidemy/forum#220] [其他問題] [answered]
### 針對Chrome DevTool中練習debugger
* 範例：project2：印出因數
* 狀況敘述如下：

　　練習時有時候直覺上就能解題我會想自己先寫看看解法，然後才反寫成虛擬碼；最後才回去對照老師的寫法與說明，之前前面可能題目根討論相對好理解；一切都滿順利的。

　　但到了練習跑debugger跟解讀時就翻車了，我知道我老是喜歡宣告變數時仍然使用var是壞習慣，但就改不太掉，因此出現 **下面圖片的狀況**。

　　**我其實也不知道要如何提問這個狀況**，當下只知道如果用var宣告變數包含`i`好像無法觀察步驟。
因為通常 `i`我都用let宣告，想說反正是區塊內；用let很合理，外面用var好像沒差，對照老師的步驟就整個看起來不同，比如根本沒看到紀錄`sctipt>步驟`的部分`。

我的狀況
![image](https://user-images.githubusercontent.com/73917475/136689776-67319a73-2b6b-4ac8-a83d-a64ecfd17241.png)

正常的狀況
![image](https://user-images.githubusercontent.com/73917475/136689818-fcd6c2b2-fb74-4a42-8bda-d0fd669dd48e.png)

```
<script>

debugger

let num = 30
for(let i=1; i<=num; i++) {
  if (num % i === 0) {
    console.log(i)
  }
}



/*
var Num = 30
var elements = []
//用let，debug時 會開頭就記錄在Scop/Script中，設值為undefined

for(let i=1; i<=Num; i++){
  if(Num % i ===0){
    elements.push(i)
    Num = Num/i
  }
}
console.log(elements)
//結果我得到因數也不是質數ToT
*/

</script>
```

**Comment by aszx87410** [comment, marked as answer]
我猜測你的 num 可能是在 global scope 裡面，你把 scope 裡的 gloabl 點開找看看

**Reply by IamTomIamQQ** [reply]
第一次看到時打開golbal覺得眼花撩亂 沒找到，剛剛仔細看；原來我又看漏了。  

**不過當時我想問的是為何都是宣告在外部 var會跑到global 而 let 卻是在Script？**  
>但直覺上總覺得我不該去追問這些我暫時無法處裡的資訊 哀哀哀 (QoQ)

**Reply by aszx87410** [reply]
因為這兩個宣告變數的方式的 scope 不一樣，想深入了解可以用 `let var scope` 作為關鍵字去研究
